Context: The insurance company wants to know more of the stock in the shop. 

Business quaestions to answer:
1. How many products contain the word Fire in their description?
2. How many products contain the word candle in their description?
3. How many products contain the word matches in their description?


In [6]:
# Import Pandas and numpy
import pandas as pd
import numpy as np

# Read the CSV file from the current working directory.
products = pd.read_csv('products.csv')

# View the DataFrame
print(products.shape)
print(products.columns)
print(products.dtypes)

(3945, 2)
Index(['StockCode', 'Description'], dtype='object')
StockCode      object
Description    object
dtype: object


In [7]:
# Create a user defined function for product descriptions containing 'fire'. 
def contains_fire(a):
    """ Does the description contain word 'fire'?"""
    b = a.lower()
    return "fire" in b
print(contains_fire)

<function contains_fire at 0x000001662E2F9EE0>


In [8]:
# Create a user defined function for product descriptions containing 'candle'. 
def contains_candle(c):
    """ Does the description contain word 'candle'?"""
    d = c.lower()
    return "candle" in d
print(contains_candle)

<function contains_candle at 0x000001662E25EF70>


In [9]:
# Create a user defined function for products descriptions containing 'matches'. 
def contains_matches(e):
    """ Does the description contain word 'matches'?"""
    f = e.lower()
    return "matches" in f
print(contains_matches)

<function contains_matches at 0x000001662E3184C0>


In [10]:
# Use the apply() function on Series.
ff = products['Description'].apply(contains_fire)

# View the DataFrame.
print(ff)

# Filter the DataFrame.
products[ff]

0       False
1       False
2       False
3       False
4       False
        ...  
3940    False
3941    False
3942    False
3943    False
3944    False
Name: Description, Length: 3945, dtype: bool


StockCode                          Description
1318     22428             ENAMEL FIRE BUCKET CREAM
3033    84569B       PACK 3 FIRE ENGINE/CAR PATCHES
3626    90039A     FIRE POLISHED GLASS BRACELET RED
3627    90039B  FIRE POLISHED GLASS BRACELET MONTAN
3628    90039C   FIRE POLISHED GLASS BRACELET BLACK
3629    90039D   FIRE POLISHED GLASS BRACELET GREEN
3656    90060B       FIRE POLISHED GLASS NECKL GOLD
3657    90060D     FIRE POLISHED GLASS NECKL BRONZE
3658    90060F      FIRE POLISHED GLASS NECKL GREEN

In [11]:
# Use the apply() function on Series.
fc = products['Description'].apply(contains_candle)

# View the DataFrame.
print(fc)

# Filter the DataFrame.
products[fc]

0       False
1       False
2       False
3       False
4       False
        ...  
3940    False
3941    False
3942    False
3943    False
3944    False
Name: Description, Length: 3945, dtype: bool


StockCode                         Description
83      17012A  ORIGAMI VANILLA INCENSE/CANDLE SET
84      17012B  ORIGAMI JASMINE INCENSE/CANDLE SET
86      17012D     ORIGAMI ROSE INCENSE/CANDLE SET
87      17012E    ORIGAMI OPIUM INCENSE/CANDLE SET
107     17136A      BLUE STONES ON WIRE FOR CANDLE
...        ...                                 ...
3530    85230A         OPIUM SCENTED VOTIVE CANDLE
3531    85230B      CINNAMON SCENTED VOTIVE CANDLE
3532    85230E      STRAWBRY SCENTED VOTIVE CANDLE
3533    85230F               JASMINE VOTIVE CANDLE
3534    85230G                ORANGE VOTIVE CANDLE

[178 rows x 2 columns]

In [12]:
# Use the apply() function on Series.
fm = products['Description'].apply(contains_matches)

# View the DataFrame.
print(fm)

# Filter the DataFrame.
products[fm]

0       False
1       False
2       False
3       False
4       False
        ...  
3940    False
3941    False
3942    False
3943    False
3944    False
Name: Description, Length: 3945, dtype: bool


StockCode                      Description
669     21584     RETROSPOT SMALL TUBE MATCHES
670     21586  KINGS CHOICE GIANT TUBE MATCHES
671     21587     COSY HOUR GIANT TUBE MATCHES
672     21588     RETROSPOT GIANT TUBE MATCHES
673     21591      COSY HOUR CIGAR BOX MATCHES
674     21592      RETROSPOT CIGAR BOX MATCHES

Answers to business questions:
1. Number of products containing word 'fire' : 9
2. Number of products containing word 'candle' : 178
3. Number of products containing word 'matches' : 6